gslb/ingestion/gslb_host_rule.go

```diff
@@ -159,21 +212,21 @@ func ValidateGSLBHostRule(gslbhr *gslbhralphav1.GSLBHostRule) error {
 	//    GSLBService.
 	sitePersistence := gslbhrSpec.SitePersistence
 	if sitePersistence != nil {
-		if sitePersistence.Enabled == true && !isSitePersistenceProfilePresent(gslbhr, sitePersistence.ProfileRef) {
-			errmsg = "SitePersistence Profile " + sitePersistence.ProfileRef + " error for " + gslbhrName + " GSLBHostRule"
+		sitePersistenceProfileName := sitePersistence.ProfileRef
+		if sitePersistence.Enabled == true && isSitePersistenceProfilePresent(gslbhr, sitePersistenceProfileName) != true {
+			errmsg = "SitePersistence Profile " + sitePersistenceProfileName + " error for " + gslbhrName + " GSLBHostRule"
+			return fmt.Errorf(errmsg)
 		}
 	}
 
 	thirdPartyMembers := gslbhrSpec.ThirdPartyMembers
 	for _, tpmember := range thirdPartyMembers {
 		if vip := net.ParseIP(tpmember.VIP); vip == nil {
 			errmsg := "Invalid VIP for thirdPartyMember site " + tpmember.Site + "," + gslbhrName + " GSLBHostRule (expecting IP address)"
-			updateGSLBHR(gslbhr, errmsg, GslbHostRuleRejected)
```

Comment:
Shouldn't we update the GSLB HR status if the member is invalid?

Reply:
Just sending an error to the event handler here. The update of GSLB HR status is taken care of in the event handlers.

Reply:
Makes sense.